src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Windows.cs

```diff
@@ -300,31 +300,15 @@ public static unsafe SocketError SendFile(SafeSocketHandle handle, SafeFileHandl
             }
         }
 
-        public static unsafe SocketError SendTo(SafeSocketHandle handle, byte[] buffer, int offset, int size, SocketFlags socketFlags, byte[] peerAddress, int peerAddressSize, out int bytesTransferred)
+        public static SocketError SendTo(SafeSocketHandle handle, byte[] buffer, int offset, int size, SocketFlags socketFlags, byte[] peerAddress, int peerAddressSize, out int bytesTransferred) =>
+            SendTo(handle, new ReadOnlySpan<byte>(buffer, offset, size), socketFlags, peerAddress, peerAddressSize, out bytesTransferred);
+
+        public static unsafe SocketError SendTo(SafeSocketHandle handle, ReadOnlySpan<byte> buffer, SocketFlags socketFlags, byte[] peerAddress, int peerAddressSize, out int bytesTransferred)
         {
             int bytesSent;
-            if (buffer.Length == 0)
+            fixed (byte* bufferPtr = &MemoryMarshal.GetReference(buffer))
```

Comment:
```suggestion
            fixed (byte* bufferPtr = buffer)
```
In the case `buffer` is empty this is guaranteed to return the null-ptr (to be precise: a Null-ref is returned which is then the null-ptr).

`MemoryMarshal.GetReference` doesn't provide that guarantee. So one should check for `IsEmpty` before. As it goes here to the kernel-call I'd go with the safe version.

Reply:
The span versions of `Send` and `Receive` [do the same thing](https://github.com/dotnet/runtime/blob/2223babdd49118787c675e04aff711f936a10b26/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Windows.cs#L278) which I copied. Should I change those too?

Reply:
Had another thought about this. As you pass `buffer.Length` to the Win-API your code is OK, and please ignore my comment here.

FYI https://github.com/dotnet/runtime/issues/25138#issuecomment-375163012 handles that case is general.